### PR TITLE
fix #413: move internal events from bounded P4 event_tx to unbounded P2 role_tx — eliminate deadlock risk

### LINKED ISSUE #413: fix: move internal events from bounded P4 event_tx to unbounded P2 role_tx — eliminate deadlock risk
## Background

The three-channel design (P2/P3/P4) is intentionally correct. This ticket
enforces its original intent by fixing four boundary violations.

---

## Move internal events off the bounded P4 channel

Six variants in `RaftEvent` (bounded `event_tx`, P4) are internal lifecycle
events, not external RPCs:

`CreateSnapshotEvent` / `LogPurgeCompleted` / `SnapshotCreated` /
`PromoteReadyLearners` / `StepDownSelfRemoved` / `MembershipApplied`

Under heavy RPC load, `event_tx` saturates. Internal tasks block at
`event_tx.send().await` while the Raft loop drains only RPC events —
e.g. Main loop waits for LogPurgeCompleted to proceed, but LogPurgeCompleted is blocked on a full event_tx (saturated by inbound RPCs). Neither side can make progress

---
## Rename RaftEvent to InboundEvent and rename RoleEvent to InternalEvent

### PR DESCRIPTION
**Pull request by JoshuaChi**
## What Does This PR Do?

Migrates 6 internal lifecycle events (`CreateSnapshotEvent`, `SnapshotCreated`, `LogPurgeCompleted`, `StepDownSelfRemoved`, `MembershipApplied`, `PromoteReadyLearners`) from the bounded `event_tx` (P4) channel to the unbounded `role_tx` (P2) channel, and renames `RaftEvent`→`InboundEvent` / `RoleEvent`→`InternalEvent` for clarity.

**Type:**
- [x] Bug Fix (with test)
- [ ] Feature (issue #\_\_\_ approved)
- [ ] Documentation
- [ ] Test/Coverage
- [ ] Performance (with benchmark)

---

## Why Is This Needed?

Fixes #413. The 6 lifecycle events were sent over the bounded `event_tx` channel. Under high inbound RPC load, `event_tx` fills up and background tasks block on `send().await` while the main loop is blocked waiting to process those same RPCs — **deadlock**.

Fix: move these internal-only events (no response channel, safe to drop on wrong role) to the unbounded `role_tx`. Follower/Learner now silently ignore stale Leader-only events instead of returning `RoleViolation`, which is the correct behavior given that `BecomeFollower` and these events now share the same channel and ordering is a race.

---

## Checklist

**Required:**
- [ ] `make test` passes
- [x] Added tests for new code
- [x] Commits squashed to 1-2 logical units

**If changing APIs:**
- [ ] Updated relevant docs
- [x] Explained why complexity is justified (see Reviewer Notes)

---

## Testing

**How tested:**
- Unit tests: `follower_state_test::test_follower_ignores_stale_leader_internal_events`, `learner_state_test::test_learner_ignores_stale_leader_internal_events`
- Unit tests: dispatch-layer tests for all 6 migrated `InternalEvent` variants in `raft.rs`
- Integration tests: N/A
- Manual testing: N/A

**For bug fixes:**
- [x] Added test that fails without this fix

---

## Does This Follow d-engine's Principles?

- [x] Solves a real problem for most users (not just my edge case)
- [x] Keeps implementation simple
- [x] Doesn't bloat the API surface

---

## Reviewer Notes

**Channel priority recap:**

```
P2 role_tx (unbounded) — now carries both BecomeFollower and the 6 lifecycle events
P4 event_tx (bounded)  — inbound RPCs only
```

Because `BecomeFollower` and these events now share P2, ordering is a race. Follower/Learner receiving `LogPurgeCompleted` / `PromoteReadyLearners` / `MembershipApplied` after step-down is normal in-flight behavior. `Candidate` retains `RoleViolation` — `role_tx` is fully drained between `Leader→Follower` and `Follower→Candidate`, so a Candidate receiving these events is a true protocol violation.

`is_fatal()` guard added to `StepDownSelfRemoved` and `LogPurgeCompleted` dispatch arms for symmetry with the other 4 arms.

See `LIFECYCLE_EVENT_MIGRATION_ANALYSIS.md` for full protocol correctness analysis including the 4 open questions (Q1–Q4).

**Estimated review complexity:**
- [ ] Quick (< 100 lines)
- [x] Medium (< 300 lines)
- [ ] Deep (> 300 lines)

## Summary by CodeRabbit

* **Refactor**
  * Streamlined the async IO event-loop architecture around inbound/internal events to maintain non-blocking behavior under write load.

* **Protocol Changes**
  * Updated `ChunkStatus` in the storage protocol: added `CHUNK_STATUS_UNSPECIFIED`, renamed existing values with a `CHUNK_STATUS_` prefix, and shifted the on-wire numeric codes.

* **Documentation**
  * Clarified snapshot generation guidance to reference the “Inbound event loop”.
  * Reformatted the server configuration reference table for readability.
  * Updated changelog/README wording to match the revised event-loop terminology.

### CODE REVIEW

File: d-engine-proto/proto/server/storage.proto [hunk 1]
```diff
@@ -50,11 +50,12 @@ message SnapshotAck {
...
-    OUT_OF_ORDER = 2;
-    REQUESTED = 3;
-    FAILED = 4;
+    CHUNK_STATUS_UNSPECIFIED = 0;
+    CHUNK_STATUS_ACCEPTED = 1;
+    CHUNK_STATUS_CHECKSUM_MISMATCH = 2;
+    CHUNK_STATUS_OUT_OF_ORDER = 3;
+    CHUNK_STATUS_REQUESTED = 4;
+    CHUNK_STATUS_FAILED = 5;
   }
```
[the remark this answers is not in the document]
Comment (JoshuaChi) [reply d-engine-proto/proto/server/storage.proto:59]: We only support atomic upgrades (all nodes upgrade simultaneously) — rolling upgrades
are out of scope for this project at v0.2.x. The zero-value UNSPECIFIED is required
by proto3 spec and intentional. Closing as won't fix.
